common/src/stack/report-system/command/report/system/tests/test_other_dhcp.py

```diff
@@ -57,7 +57,27 @@ def test_other_dhcp_server(interface):
 	# Set up socket with passed in interface
 	send_dhcp = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
 	send_dhcp.setsockopt(socket.SOL_SOCKET, socket.SO_BROADCAST, 1)
-	send_dhcp.setsockopt(socket.SOL_SOCKET, 25, str(interface + '\0').encode('utf-8'))
+
+	# Bail if we can't setup a socket on the current interface
+	try:
+		send_dhcp.setsockopt(socket.SOL_SOCKET, 25, str(interface + '\0').encode('utf-8'))
+
+	except OSError:
+		pytest.skip(f'Could not setup socket for {interface}')
+
+	# If the MAC address isn't in the Stacki database, try elsewhere
+	if not pxe_interfaces[interface]:
+		find_mac = _exec(['cat'] + [f'/sys/class/net/{interface}/address'])
+		stdout, stderr = (find_mac.stdout, find_mac.stderr)
+
+		# If we can't get the address from /sys/class/net, skip this interface
+		if stderr:
+			pytest.skip(f'Could not find mac address for {interface}')
+		else:
+			mac_addr = stdout.strip()
+
+	else:
+		mac_addr = pxe_interfaces[interface]
```

Comment:
The number of cases could be reduced by doing something more like the following:
```
mac_addr = pxe_interfaces[interface]
if not mac_addr:
    ...
```

The inner `if stderr:` doesn't need an `else` either as the `skip` will exit the function, I believe anyways.

Reply:
I ran it on a virtual frontend using report system and made sure it didn't stacktrace like before when no MAC was in the database.